modules/interacting-serverless-apps-http2-gRPC.adoc

```diff
@@ -3,76 +3,43 @@
 
 [IMPORTANT]
 ====
-This method needs to expose Kourier Gateway using the `LoadBalancer` service type. You can configure this by adding the following YAML to your `KnativeServing` custom resource definition (CRD):
-
-[source,yaml]
-----
-...
-spec:
-  ingress:
-    kourier:
-      service-type: LoadBalancer
-...
-----
+This method applies to {product-title} 4.10 and later. For older versions, see the following section.
 ====
 
 .Prerequisites
 
 * {ServerlessOperatorName} and Knative Serving are installed on your cluster.
 * Install the OpenShift CLI (`oc`).
 * You have created a Knative service.
+* {product-title} 4.10 or later.
+* You have enabled HTTP/2 on OpenShift Ingress controller.
 
 .Procedure
 
-. Find the application host. See the instructions in _Verifying your serverless application deployment_.
-
-. Find the ingress gateway's public address:
-+
-[source,terminal]
-----
-$ oc -n knative-serving-ingress get svc kourier
-----
-+
-.Example output
+. Add the `serverless.openshift.io/default-enable-http2=true` annotation to the `KnativeServing` Custom Resource:
 +
 [source,terminal]
 ----
-NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP                                                             PORT(S)                                                                                                                                      AGE
-kourier   LoadBalancer   172.30.51.103   a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com   80:31380/TCP,443:31390/TCP   67m
+$ oc annotate knativeserving <your_knative_CR> -n knative-serving serverless.openshift.io/default-enable-http2=true
 ----
-+
-The public address is surfaced in the `EXTERNAL-IP` field, and in this case is `a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com`.
 
-. Manually set the host header of your HTTP request to the application's host, but direct the request itself against the public address of the ingress gateway.
+. After the annotation is added, you can verify that the `appProtocol` value of the Kourier service is `h2c`:
 +
 [source,terminal]
 ----
-$ curl -H "Host: hello-default.example.com" a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com
+$ oc get svc -n knative-serving-ingress kourier -o jsonpath="{.spec.ports[0].appProtocol}"
 ----
 +
 .Example output
-[source,terminal]
-----
-Hello Serverless!
-----
 +
-You can also make a gRPC request by setting the authority to the application's host, while directing the request against the ingress gateway directly:
-+
-[source,yaml]
+[source,terminal]
 ----
-grpc.Dial(
-    "a83e86291bcdd11e993af02b7a65e514-33544245.us-east-1.elb.amazonaws.com:80",
-    grpc.WithAuthority("hello-default.example.com:80"),
-    grpc.WithInsecure(),
-)
+h2c
 ----
 +
-[NOTE]
-====
-Ensure that you append the respective port, 80 by default, to both hosts as shown in the previous example.
-====
+Now you can use the gRPC framework over the HTTP/2 protocol for external traffic.
```

Comment:
Does it make sense to include the `curl` example here, similar to what we have for older OpenShift? By "gRPC framework" you probably mean the "google.golang.org/grpc" library (as mentioned in my previous comment). If we keep the Golang example for OSD 4.9 then we could at least point to it from here so  that users know what is "gRPC framework"

Reply:
When we test gRPC app, we cannot use `curl` but we need to `grpcurl`, some other tools. And the `grpcurl` does is not an official Red Hat product. So I think we cannot add the `curl` or some handy example.

Let's add the `Golang example` as a code snippet here just for now. Ideally I think we should add an sample gRPC application in https://github.com/openshift-knative/showcase with the gRPC client code.